### ISSUE TITLE
Uploading to root folder of a bucket

### ISSUE THREAD

**Issue by RichardForshaw** [bradleyg/django-s3direct#46]
Hi,

The code currently marks an empty path in S3DIRECT_DESTINATIONS as an error condition, and also does not handle the case for specifying '/' very well. I think that both of these are valid cases where the user is specifying that the file be uploaded to the root folder of the bucket. The only way to do this seems to be to provide a function which just returns '${filename}'

I suggest changing views.py to something like:

```python
    if hasattr(key, '__call__'):
        key = key(filename)
    elif key and key !='/':
        key = '%s/${filename}' % key
    else:
        #Key is empty or is '/' - assume the request is to put in the root of the bucket
        key = '${filename}'
```

**Comment by bradleyg** [comment]
This makes total sense, and I'm surprised hasn't been mentioned before. Any chance you could test and make a pull request? :)

**Comment by RichardForshaw** [comment]
Yes I will try and get something ready. I also have some other suggestions - keen to help if I can!

**Comment by bradleyg** [comment]
Fixed in ecad9b8a8f3409a065d338b62b2f4a13ebb9b8de
